Mathlib/Analysis/InnerProductSpace/GramSchmidtOrtho.lean

```diff
@@ -46,25 +46,24 @@ namespace InnerProductSpace
 /-- The Gram-Schmidt process takes a set of vectors as input
 and outputs a set of orthogonal vectors which have the same span. -/
 noncomputable def gramSchmidt [WellFoundedLT ι] (f : ι → E) (n : ι) : E :=
-  f n - ∑ i : Iio n, (𝕜 ∙ gramSchmidt f i).orthogonalProjection (f n)
+  f n - ∑ i : Iio n, (𝕜 ∙ gramSchmidt f i).starProjection (f n)
 termination_by n
 decreasing_by exact mem_Iio.1 i.2
 
 /-- This lemma uses `∑ i in` instead of `∑ i :`. -/
 theorem gramSchmidt_def (f : ι → E) (n : ι) :
-    gramSchmidt 𝕜 f n = f n - ∑ i ∈ Iio n, (𝕜 ∙ gramSchmidt 𝕜 f i).orthogonalProjection (f n) := by
+    gramSchmidt 𝕜 f n = f n - ∑ i ∈ Iio n, (𝕜 ∙ gramSchmidt 𝕜 f i).starProjection (f n) := by
   rw [← sum_attach, attach_eq_univ, gramSchmidt]
 
 theorem gramSchmidt_def' (f : ι → E) (n : ι) :
-    f n = gramSchmidt 𝕜 f n + ∑ i ∈ Iio n, (𝕜 ∙ gramSchmidt 𝕜 f i).orthogonalProjection (f n) := by
+    f n = gramSchmidt 𝕜 f n + ∑ i ∈ Iio n, (𝕜 ∙ gramSchmidt 𝕜 f i).starProjection (f n) := by
   rw [gramSchmidt_def, sub_add_cancel]
 
--- changing the definition to use `starProjection` makes the proof of this not work
 theorem gramSchmidt_def'' (f : ι → E) (n : ι) :
     f n = gramSchmidt 𝕜 f n + ∑ i ∈ Iio n,
       (⟪gramSchmidt 𝕜 f i, f n⟫ / (‖gramSchmidt 𝕜 f i‖ : 𝕜) ^ 2) • gramSchmidt 𝕜 f i := by
-  convert gramSchmidt_def' 𝕜 f n
-  rw [← starProjection_apply, starProjection_singleton, RCLike.ofReal_pow]
+  nth_rw 1 [gramSchmidt_def' 𝕜 f n]
+  simp only [starProjection_singleton, map_pow]
```

Comment:
I don't understand the `nth_rw`. Doesn't the `n` mean that it matches only the first term on the RHS? I would expect this to work (untested):
```suggestion
  simp [gramSchmidt_def' 𝕜 f n, starProjection_singleton]
```
unless there is some non-confluence in our `simp` set somewhere.